File: src/main/java/com/example/leaderboard/domain/user/repository/UserRepository.java
package com.example.leaderboard.domain.user.repository;

import com.example.leaderboard.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserRepository extends JpaRepository<User, Long> {
    boolean existsByUsername(String username);
    boolean existsByEmail(String email);
    List<User> findAllByIdIn(List<Long> ids);
}

File: src/main/kotlin/com/example/leaderboard/domain/user/repository/UserRepository.kt
package com.example.leaderboard.domain.user.repository

import com.example.leaderboard.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    fun existsByUsername(username: String): Boolean
    fun existsByEmail(email: String): Boolean
    fun findAllByIdIn(ids: List<Long>): List<User>
}
